dkg/runner.go

```diff
@@ -132,15 +149,70 @@ func (r *Runner) ProcessMsg(msg *SignedMessage) (bool, map[types.OperatorID]*Sig
 }
 
 func (r *Runner) signAndBroadcastMsg(msg types.Encoder, msgType MsgType) error {
-	panic("implement")
+	data, err := msg.Encode()
+	if err != nil {
+		return err
+	}
+	signedMessage := &SignedMessage{
+		Message: &Message{
+			MsgType:    msgType,
+			Identifier: r.Identifier,
+			Data:       data,
+		},
+		Signer:    r.Operator.OperatorID,
+		Signature: nil,
+	}
+	// GLNOTE: Should we use SignDKGOutput?
+	sig, err := r.config.Signer.SignDKGOutput(signedMessage, r.Operator.ETHAddress)
```

Comment:
that not what this function is about, it takes an encoder interface and broadcast it to the network.. it's not a msg specific func

Reply:
I'm confused. The function name says we need to sign the message. But if we don't include the `SignedMessage` and `SignDKGOutput`, how do we do the signing?